crates/evm/src/executor/inspector/cheatcodes/util.rs

```diff
@@ -171,14 +167,14 @@ pub fn check_if_fixed_gas_limit<DB: DatabaseExt>(
     // time of the call, which should be rather close to configured gas limit.
     // TODO: Find a way to reliably make this determination. (for example by
     // generating it in the compilation or evm simulation process)
-    U256::from(data.env.tx.gas_limit) > data.env.block.gas_limit.to_ethers() &&
-        U256::from(call_gas_limit) <= data.env.block.gas_limit.to_ethers()
+    U256::from(data.env.tx.gas_limit) > data.env.block.gas_limit &&
+        U256::from(call_gas_limit) <= data.env.block.gas_limit
         // Transfers in forge scripts seem to be estimated at 2300 by revm leading to "Intrinsic
         // gas too low" failure when simulated on chain
         && call_gas_limit > 2300
 }
 
 /// Small utility function that checks if an address is a potential precompile.
-pub fn is_potential_precompile(address: H160) -> bool {
-    address < H160::from_low_u64_be(10) && address != H160::zero()
+pub fn is_potential_precompile(address: Address) -> bool {
+    address < Address::from_word(B256::from(U256::from(10))) && address != Address::ZERO
```

Comment:
`Address::from_word(B256::from(U256::from(10)))` is this the intended api? looks terrible, could we do something better? isn't `address!(0x11)` working here?

Reply:
`Address::with_last_byte(10)`

`FixedBytes <-> Integers` is weird since it's `FixedBytes` generic so a misuse would be a runtime panic. We would have to implement these conversions for each fixed byte and integer pair with both a big and little endian method